modules/ximgproc/include/opencv2/ximgproc/fourier_descriptors.hpp

```diff
@@ -0,0 +1,106 @@
+            std::vector<double> rho, psi;
+            void frequencyInit();
+            void fAlpha(double x, double &fn, double &df);
+            double distance(std::complex<double> r, double alpha);
+            double  newtonRaphson(double x1, double x2);
+        public:
+            /** @brief Fit two closed curves using fourier descriptors. More details in @cite PersoonFu1977 and @cite BergerRaghunathan1998
+
+            @param ctr number of Fourier descriptors equal to number of contour points after resampling.
+            @param fd Contour defining second shape (Target).
+            @example fourier_descriptors_demo.cpp
+            */
+            ContourFitting(int ctr=1024,int fd=16):ctrSize(ctr),fdSize(fd){};
+            /** @brief Fit two closed curves using fourier descriptors. More details in @cite PersoonFu1977 and @cite BergerRaghunathan1998
+
+            @param src Contour defining first shape.
+            @param dst Contour defining second shape (Target).
+            @param _alphaPhiST : \f$ \alpha \f$=_alphaPhiST(0,0), \f$ \phi \f$=_alphaPhiST(0,1) (in radian), s=_alphaPhiST(0,2), Tx=_alphaPhiST(0,3), Ty=_alphaPhiST(0,4) rotation center
+            @param dist distance between src and dst after matching.
+            @param fdContour false then src and dst are contours and true src and dst are fourier descriptors.
+            @example fourier_descriptors_demo.cpp
+            */
+            void estimateTransformation(InputArray src, InputArray dst, OutputArray _alphaPhiST, double *dist=0,bool fdContour=false);
+            /** @brief set number of Fourier descriptors used in estimateTransformation
+
+            @param n number of Fourier descriptors equal to number of contour points after resampling.
+            */
+            void setCtrSize(int n);
+            /** @brief set number of Fourier descriptors when estimateTransformation used vector<Point>
+
+            @param n number of fourier descriptors used for optimal curve matching.
+            */
+            void setFDSize(int n);
+            /**
+            @returns number of fourier descriptors
+            */
+            int getCtrSize() { return ctrSize; };
+            /**
+            @returns number of fourier descriptors used for optimal curve matching
+            */
+            int getFDSize() { return fdSize; };
+        };
+        /**
+        * @brief   Fourier descriptors for planed closed curves
+        *
+        * For more details about this implementation, please see @cite PersoonFu1977
+        *
+        * @param   _src   contour type vector<Point> , vector<Point2f>  or vector<Point2d>
+        * @param   _dst   Mat of type CV_64FC2 and nbElt rows A VERIFIER
+        * @param   nbElt number of rows in _dst or getOptimalDFTSize rows if nbElt=-1
+        * @param   nbFD number of FD return in _dst _dst = [FD(1...nbFD/2) FD(nbFD/2-nbElt+1...:nbElt)]
+        *
+        * @example fourier_descriptors_demo.cpp
```

Comment:
You can get rid of these `example` commands and leave just one mention of `fourier_descriptors_demo.cpp` somewhere.